src/xdsl/xdsl_opt_main.py

```diff
@@ -63,13 +66,33 @@ def run(self):
         Executes the different steps.
         """
         module = self.parse_input()
-        if not self.args.verify_diagnostics:
+        if self.args.verbose_trace:
+            print("a")
+            try:
+                self.apply_passes(module)
+
+            except Exception as e:
+                # in case there is an error in the formatter
+                try:
+                    frame_count = self.args.verbose_trace[0]
+                    debug_str = verbose(e, frame_count)
+                    print(debug_str, file=sys.stderr)
+
+                except Exception:
+                    print(e)
+
+                exit(0)
+
+        elif not self.args.verify_diagnostics:
             self.apply_passes(module)
         else:
             try:
                 self.apply_passes(module)
+
```

Comment:
Could we remove those changes from this PR? This makes the blame quite unreadable.